src/components/AdvocatesTable/index.tsx

```diff
@@ -0,0 +1,113 @@
+"use client";
+
+import { AdvocateResultsT } from "@/app/api/advocates/route";
+import {
+  flexRender,
+  getCoreRowModel,
+  useReactTable,
+} from "@tanstack/react-table";
+
+import { Button } from "@/components/ui/button";
+
+import {
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from "@/components/ui/table";
+
+import { columns } from "./columns";
+
+interface Props {
+  advocateData: AdvocateResultsT["data"];
+  onPageChange: (page: number) => void;
+}
+
+export function AdvocatesTable({ advocateData, onPageChange }: Props) {
```

Comment:
When/if we had another index view that utilized a similar table, I'd abstract this into a more generic component when a reusable pattern is more clear.